pandas/io/parsers.py

```diff
@@ -2327,14 +2327,21 @@ def _next_line(self):
                 try:
                     orig_line = next(self.data)
                 except csv.Error as e:
+                    msg = str(e)
+
                     if 'NULL byte' in str(e):
-                        raise csv.Error(
-                            'NULL byte detected. This byte '
-                            'cannot be processed in Python\'s '
-                            'native csv library at the moment, '
-                            'so please pass in engine=\'c\' instead.')
-                    else:
-                        raise
+                        msg = ('NULL byte detected. This byte '
+                               'cannot be processed in Python\'s '
+                               'native csv library at the moment, '
+                               'so please pass in engine=\'c\' instead')
+
+                    if self.skipfooter > 0:
+                        reason = ('Error could possibly be due to '
+                                  'skipfooter being ignored in Python\'s '
+                                  'native csv library.')
```

Comment:
Trying a long version:

> Error could possibly be due to parsing errors in the skipped footer rows (the `skipfooter` keyword is only applied after Python's csv library has parsed all rows)

(is that correct?) 
If you have ideas to shorten it a bit but keep it clear, welcome :-)